src/class-main.php

```diff
@@ -38,6 +37,13 @@
  */
 class Main extends Lib_Core {
 
+  /**
+   * Set this to true if you need dependency injection in tests.
+   *
+   * @var boolean
+   */
+  protected $is_test = false;
```

Comment:
If this is used only in this class, it can be set to `private`. I generally prefer using either defined constants or env variables for this. Env variables are great because you can override them in tests (if using wp-browser).

Reply:
I'd say implementation details, you can just call up $main->set_test(true / false) in your `BaseTest` class as needed (for example WPBrowser tests could extend `BaseWPBrowserTest` class which just runs `$main->set_test(true)` in it's `_inject()` or wherever if needed).